clang/test/Driver/entry.c

```diff
@@ -0,0 +1,5 @@
+// RUN: %clang -### --entry test %s 2>&1 | FileCheck %s
```

Comment:
In the absence of `--target=`, the default target triple `LLVM_DEFAULT_TARGET_TRIPLE` is used. The selected `ToolChain` may not run AddLinkerInputs to render `-e`.

We should add `--target=` (common choice is a linux triple) and try reusing an existing test https://maskray.me/blog/2021-08-08-toolchain-testing#i-dont-know-an-existing-test-can-be-enhanced

Perhaps linker-opts.c or Xlinker-args.c